lib/arpv4/arpv4.ml

```diff
@@ -73,21 +73,24 @@ module Make (Ethif : V1_LWT.ETHIF) (Clock : V1.CLOCK) (Time : V1_LWT.TIME) = str
     Format.fprintf fmt "%s" repr
 
   let notify t ip mac =
-    let now = Clock.time () in
-    let expire = now +. arp_timeout in
-    try
-      match Hashtbl.find t.cache ip with
-      | Pending (_, w) ->
-        Hashtbl.replace t.cache ip (Confirmed (expire, mac));
-        Lwt.wakeup w (`Ok mac)
-      | Confirmed _ ->
+    Log.debug (fun f -> f "notifying: %a -> %s" Ipaddr.V4.pp_hum ip (Macaddr.to_string mac));
+    match Ipaddr.V4.is_multicast ip || (Ipaddr.V4.compare ip Ipaddr.V4.any = 0) with
```

Comment:
http://www.ietf.org/rfc/rfc894.txt pretty heavily implies that ARP is only reasonable to use with unicast IPs.  I agree that it would be good to also filter out broadcast but the current API makes that annoying; I'd like to include that in future improvements, after resolving https://github.com/mirage/mirage/issues/557 .